src/linear_algebra/linear_pmap.lean

```diff
@@ -419,3 +419,34 @@ def coprod (f : linear_pmap R E G) (g : linear_pmap R F G) :
 rfl
 
 end linear_pmap
+
+/-! ### Graph -/
+section graph
+
+namespace linear_pmap
+
+/-- The graph of a `linear_pmap` viewed as a submodule on `E × F`. -/
+def graph (f : linear_pmap R E F) : submodule R (E × F) :=
+f.to_fun.graph.map (f.domain.subtype.prod_map linear_map.id)
+
+lemma mem_graph_iff' (f : linear_pmap R E F) {x : E × F} :
+  x ∈ graph f ↔ ∃ y : f.domain, (↑y, f y) = x :=
+by { simp [graph] }
+
+@[simp] lemma mem_graph_iff (f : linear_pmap R E F) {x : E × F} :
+  x ∈ graph f ↔ ∃ y : f.domain, (↑y : E) = x.fst ∧ f y = x.snd :=
+by { cases x, simp_rw [mem_graph_iff', prod.mk.inj_iff] }
+
+/-- The property that `f 0 = 0` in terms of the graph. -/
+lemma graph_fst_eq_zero_snd (f : linear_pmap R E F) {x : E × F} (hx : x ∈ graph f)
+  (hx' : x.fst = 0) :
+  x.snd = 0 :=
```

Comment:
Could you please add these lemmas?

- if `x y ∈ graph f` and `x.1 = y.1`, then `x = y`;
- if `(x, y) ∈ graph f` and `(x', y') ∈ graph f`, then `x = x'` implies `y = y'`;
- if `x : f.domain`, then `((x : E), f x) ∈ graph f`;
- `0 ∈ graph f`.

Reply:
done. I hope the names are fine.